classes/issue/Collector.inc.php

```diff
@@ -0,0 +1,460 @@
+        }
+
+        // Years
+        if (!is_null($this->years)) {
+            $q->whereIn('i.year', $this->years);
+        }
+
+        // Issue ids
+        if (!empty($this->issueIds)) {
+            $q->whereIn('i.issue_id', $this->issueIds);
+        }
+
+        // URL path
+        if (!empty($this->urlPath)) {
+            $q->where('i.url_path', '=', $this->urlPath);
+        }
+
+        // Search phrase
+        if (!empty($this->searchPhrase)) {
+            $searchPhrase = $this->searchPhrase;
+
+            // Add support for searching for the volume, number and year
+            // using the localized issue identification formats. In
+            // en_US this will match Vol. 1. No. 1 (2018) against:
+            // i.volume = 1 AND i.number = 1 AND i.year = 2018
+            $volume = '';
+            $number = '';
+            $year = '';
+            $volumeRegex = '/' . preg_quote(__('issue.vol')) . '\s\S/';
+            preg_match($volumeRegex, $searchPhrase, $matches);
+            if (count($matches)) {
+                $volume = trim(str_replace(__('issue.vol'), '', $matches[0]));
+                $searchPhrase = str_replace($matches[0], '', $searchPhrase);
+            }
+            $numberRegex = '/' . preg_quote(__('issue.no')) . '\s\S/';
+            preg_match($numberRegex, $searchPhrase, $matches);
+            if (count($matches)) {
+                $number = trim(str_replace(__('issue.no'), '', $matches[0]));
+                $searchPhrase = str_replace($matches[0], '', $searchPhrase);
+            }
+            preg_match('/\(\d{4}\)\:?/', $searchPhrase, $matches);
+            if (count($matches)) {
+                $year = substr($matches[0], 1, 4);
+                $searchPhrase = str_replace($matches[0], '', $searchPhrase);
+            }
+            if ($volume !== '' || $number !== '' || $year !== '') {
+                $q->where(function ($q) use ($volume, $number, $year) {
+                    if ($volume) {
+                        $q->where('i.volume', '=', $volume);
+                    }
+                    if ($number) {
+                        $q->where('i.number', '=', $number);
+                    }
+                    if ($year) {
+                        $q->where('i.year', '=', $year);
+                    }
+                });
+            }
+
+            $words = array_unique(explode(' ', $searchPhrase));
+            if (count($words)) {
+                foreach ($words as $word) {
+                    $word = strtolower(addcslashes($word, '%_'));
+                    $q->whereIn('i.issue_id', function (Builder $q) use ($word) {
+                        $q->
+                        $q->select('iss.issue_id')
+                            ->from($this->dao->settingsTable, 'iss_t')
+                            ->where('iss_t.setting_name', '=', 'title')
+                            ->where(DB::raw('lower(iss_t.setting_value'), 'LIKE', "%${word}%");
```

Comment:
Need to call `addcslashes` to escape `_` and `%` special characters that might appear in the supplied keywords. See e.g.: https://github.com/asmecher/pkp-lib/blob/7b06a009eb7db9e0d7df80d19a93b91e73462025/classes/user/Collector.inc.php#L219

Reply:
Oops, I see you've already done that above!